contracts/modules/cover/CoverViewer.sol

```diff
@@ -2,200 +2,88 @@
 
 pragma solidity ^0.8.9;
 
-import "solmate/src/tokens/ERC721.sol";
-import "@openzeppelin/contracts-v4/token/ERC20/IERC20.sol";
-import "@openzeppelin/contracts-v4/token/ERC20/utils/SafeERC20.sol";
-import "@openzeppelin/contracts-v4/proxy/beacon/UpgradeableBeacon.sol";
-
 import "../../interfaces/ICover.sol";
-import "../../interfaces/IPool.sol";
-import "../../interfaces/IERC20Detailed.sol";
 import "../../interfaces/INXMMaster.sol";
 
 contract CoverViewer {
-  struct CoverView {
-    uint24 productId;
-    uint16 productType;
-    address yieldTokenAddress;
-    uint96 amountPaidOut;
-    uint amountRemaining;
-    uint coverStart;
-    uint coverEnd;
-    uint8 coverAsset;
-    string coverAssetSymbol;
-    uint8 claimMethod;
-    uint32 gracePeriod;
+
+  struct Segment {
```

Comment:
The ICover.CoverSegment also contains:

  uint24 globalRewardsRatio;
  uint24 globalCapacityRatio;
  
  I don't se an immediate use case for it now, something to consider adding here

Reply:
Those are for internal use, not really needed outside of the contract. Since it's the CoverViewer we're talking about here we can add any time as it doesn't require a gov vote to upgrade.